mongorestore/metadata.go

```diff
@@ -88,6 +88,9 @@ func (restore *MongoRestore) LoadIndexesFromBSON() error {
 			if !bsonSource.Next(&indexDocument) {
 				break
 			}
+			if restore.OutputOptions.FixDottedHashedIndexes {
```

Comment:
This looks superfluous to me since we also do the fix in restore.RestoreIntent(). Is it necessary to have this here?

Reply:
I might be misunderstanding something here, but this code path doesn't seem to execute when indexes are loaded from a json meta data file. I'll look into when this method is actually used.

Reply:
It looks to me like we have to different ways to restore. Either from BSON or the json metadata file. I'm not sure how mongodump ever creates indexes in a BSON file, so I'm unsure how to test that. But removing this line seems like a bad idea.

Reply:
Oh, I see what you mean, now!

Reply:
I think it's from mongodb versions <3.0 which had system.indexes collections. If there are no metadata files, RestoreIntent() uses the indexes from system.indexes instead. So I think doing the fix inRestoreIntent() is sufficient since the indexes read from BSON files end up there anyway. 

We may want to remove LoadIndexesFromBSON() at some point to clean things up since we no longer support 2.6.

Reply:
You are correct. I looked through the whole code path, and even if we do restore from BSON it will still go through that other code path that fixes the indexes :)